File: src/money.js
var money= (function(){
	function money(v,curr) {

		if(v != null && curr != null && v > 0 &&  curr.length <= 3){
			this.v=v;
			this.curr=curr.toUpperCase();
		}else if( v === null || curr=== null || v < 0 ){

			throw new MoneyNullValueOrNullCurrency();
		}else if(curr !== undefined && curr.length > 3){

			throw new CurrencySupTrois();
		}else {
			this.v=null;
			this.curr=null;
		}


		this._equals = function(otherM){
			return (otherM.getValue()===this.getValue() && otherM.getCurrency().toUpperCase()===this.getCurrency().toUpperCase()) ;
		}
	}

	money.prototype.getCurrency=function () {
		return this.curr.toUpperCase();
	}
	money.prototype.getValue=function () {
		return this.v;
	}
	money.prototype.equals=function (otherM) {
		return (otherM.getValue()===this.getValue() && otherM.getCurrency().toUpperCase()===this.getCurrency().toUpperCase()) ;
	}
	money.prototype.toString=function() {
		return this.getValue()+" ("+this.getCurrency()+")" ;
	}


return money;
})();

File: Money/src/fr/ulille1/fil/odeva/Money.java
package  fr.ulille1.fil.odeva;

import java.util.HashMap;
import java.util.Map;

public class Money {
	private int value;
	private String currency;
	

	Money(int value, String currency) 
	{
		if(value <0)
				throw new InferieurZeroException();
		else
			this.value=value;
		this.currency=currency;
	}

	public int getValue()
	{
		return this.value;
	}

	public String getCurrency()
	{
		return this.currency;
	}

    public String toString() {
     	return this.getValue()+" ("+this.getCurrency()+")";
    }
    
    public boolean _equals(Object o)throws NullPointerException{
    	
    	if(((Money)o).getValue() == this.getValue() && ((Money)o).getCurrency().equals(this.getCurrency()))
    		return true;
    	return false;
    }
    
    public boolean equals(Object o){
    	return _equals(o);
    }

}
